can't build with default rustc master setup

We used to be able to
```
rustup-toolchain-install-master -n master  -f
cargo +master build
```
but this no longe works due to minimized std I think: https://github.com/rust-lang/rust/pull/64823

When I try the procedure that previously worked, I just get

```
   Compiling clippy_lints v0.0.212 (/home/matthias/vcs/github/rust-clippy/clippy_lints)
error[E0463]: can't find crate for `fmt_macros`
  --> clippy_lints/src/lib.rs:20:1
   |
20 | extern crate fmt_macros;
   | ^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error
```

We probably need to update at least the docs and CI...

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -147,7 +147,7 @@ You can use [rustup-toolchain-install-master][rtim] to do that:
 
 ```bash
 cargo install rustup-toolchain-install-master
-rustup-toolchain-install-master -n master --force
+rustup-toolchain-install-master --force -n master -c rustc-dev
 rustup override set master
 cargo test
 ```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -18,7 +18,7 @@ install:
     - rustup-init.exe -y --default-host %TARGET% --default-toolchain nightly
     - set PATH=%PATH%;C:\Users\appveyor\.cargo\bin
     - del rust-toolchain
-    - cargo install rustup-toolchain-install-master --debug || echo "rustup-toolchain-install-master already installed"
+    - cargo install --git https://github.com/kennytm/rustup-toolchain-install-master --debug || echo "rustup-toolchain-install-master already installed"
     - rustup-toolchain-install-master -f -n master -c rustc-dev
     - rustup component add rustfmt --toolchain nightly & exit 0 # Format test handles missing rustfmt
     - rustup default master

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -17,11 +17,9 @@ install:
     - curl -sSf -o rustup-init.exe https://win.rustup.rs/
     - rustup-init.exe -y --default-host %TARGET% --default-toolchain nightly
     - set PATH=%PATH%;C:\Users\appveyor\.cargo\bin
-    - git ls-remote https://github.com/rust-lang/rust.git master | awk '{print $1}' >rustc-hash.txt
-    - set /p RUSTC_HASH=<rustc-hash.txt
     - del rust-toolchain
     - cargo install rustup-toolchain-install-master --debug || echo "rustup-toolchain-install-master already installed"
-    - rustup-toolchain-install-master %RUSTC_HASH% -f -n master
+    - rustup-toolchain-install-master -f -n master -c rustc-dev
     - rustup component add rustfmt --toolchain nightly & exit 0 # Format test handles missing rustfmt
     - rustup default master
     - set PATH=%PATH%;C:\Users\appveyor\.rustup\toolchains\master\bin

--- a/setup-toolchain.sh
+++ b/setup-toolchain.sh
@@ -1,12 +1,11 @@
 #!/bin/bash
 # Set up the appropriate rustc toolchain
 
-cd "$(dirname "$0")"
+cd "$(dirname "$0")" || exit
 
 if ! command -v rustup-toolchain-install-master > /dev/null; then
   cargo install rustup-toolchain-install-master --debug
 fi
 
-RUSTC_HASH=$(git ls-remote https://github.com/rust-lang/rust.git master | awk '{print $1}')
-rustup-toolchain-install-master -f -n master "$RUSTC_HASH"
+rustup-toolchain-install-master -f -n master -c rustc-dev
 rustup override set master

--- a/setup-toolchain.sh
+++ b/setup-toolchain.sh
@@ -4,7 +4,10 @@
 cd "$(dirname "$0")" || exit
 
 if ! command -v rustup-toolchain-install-master > /dev/null; then
-  cargo install rustup-toolchain-install-master --debug
+  cargo install \
+    --git https://github.com/kennytm/rustup-toolchain-install-master \
+    --bin rustup-toolchain-install-master \
+    --debug
 fi
 
 rustup-toolchain-install-master -f -n master -c rustc-dev

